### PR TITLE
#KTOR-936 Migrate to explicit api mode(part 2)

### PR DESCRIPTION
**Pull request by e5l**
Fixes [`#KTOR-936`](https://youtrack.jetbrains.com/issue/KTOR-936)

### CODE REVIEW

File: ktor-utils/common/src/io/ktor/util/CaseInsensitiveMap.kt [hunk 1]
```diff
@@ -41,7 +41,7 @@ class CaseInsensitiveMap<Value> : MutableMap<String, Value> {
         )
 
     override val entries: MutableSet<MutableMap.MutableEntry<String, Value>>
-        get() = DelegatingMutableSet<MutableMap.MutableEntry<CaseInsensitiveString, Value>, MutableMap.MutableEntry<String, Value>>(
+        get() = DelegatingMutableSet(
```
Comment (Stexxe) [inline ktor-utils/common/src/io/ktor/util/CaseInsensitiveMap.kt:44]: How this change relates to explicit API mode?

File: ktor-utils/common/src/io/ktor/util/Crypto.kt [hunk 2]
```diff
@@ -68,40 +68,41 @@ fun generateNonce(size: Int): ByteArray = buildPacket {
  * Compute SHA-1 hash for the specified [bytes]
  */
 @KtorExperimentalAPI
-expect fun sha1(bytes: ByteArray): ByteArray
+public expect fun sha1(bytes: ByteArray): ByteArray
 
 /**
  * Create [Digest] from specified hash [name].
  */
+@Suppress("FunctionName")
```
Comment (Stexxe) [inline ktor-utils/common/src/io/ktor/util/Crypto.kt:76]: This change doesn't relate to the original issue

File: ktor-utils/common/src/io/ktor/util/Encoders.kt [hunk 3]
```diff
@@ -9,9 +9,9 @@ import io.ktor.utils.io.*
 
 @KtorExperimentalAPI
 /**
- * Empty [Encoder]
+ * Empty [Encoder].
```
Comment (Stexxe) [inline ktor-utils/common/src/io/ktor/util/Encoders.kt:12]: This change doesn't relate to the original issue

File: ktor-utils/common/src/io/ktor/util/Hash.kt [hunk 4]
```diff
@@ -2,9 +2,11 @@
  * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
  */
 
+@file:Suppress("KDocMissingDocumentation")
```
Comment (Stexxe) [inline ktor-utils/common/src/io/ktor/util/Hash.kt:5]: This change doesn't relate to the original issue

File: ktor-utils/common/src/io/ktor/util/PlatformUtils.kt [hunk 5]
```diff
@@ -4,10 +4,11 @@
 
 package io.ktor.util
 
+@Suppress("KDocMissingDocumentation")
```
Comment (Stexxe) [inline ktor-utils/common/src/io/ktor/util/PlatformUtils.kt:7]: This change doesn't relate to the original issue

File: ktor-utils/common/src/io/ktor/util/Ranges.kt [hunk 6]
```diff
@@ -8,11 +8,11 @@ package io.ktor.util
...
-val LongRange.length: Long
+public val LongRange.length: Long
     get() = (endInclusive - start + 1).coerceAtLeast(0L)
 
 /**
  * Returns `true` if [other] range is fully contained inside [this] range
  */
 @InternalAPI
-operator fun LongRange.contains(other: LongRange) = other.start >= start && other.endInclusive <= endInclusive
+public operator fun LongRange.contains(other: LongRange): Boolean = other.first >= start && other.last <= endInclusive
```
Comment (Stexxe) [inline ktor-utils/common/src/io/ktor/util/Ranges.kt:18]: Some changes don't relate to the original issue

File: ktor-utils/common/src/io/ktor/util/StringValues.kt [hunk 7]
```diff
@@ -92,9 +93,9 @@ open class StringValuesSingleImpl(
 
     override fun names(): Set<String> = setOf(name)
 
-    override fun toString() = "StringValues(case=${!caseInsensitiveName}) ${entries()}"
+    override fun toString(): String = "StringValues(case=${!caseInsensitiveName}) ${entries()}"
```
Comment (Stexxe) [inline ktor-utils/common/src/io/ktor/util/StringValues.kt:96]: This change doesn't relate to the original issue

**Review by Stexxe — CHANGES_REQUESTED** [review]
Some changes don't relate to the original problem. I left comments about a few of them.
PR has no link to the Youtrack issue in the description.

File: ktor-utils/common/src/io/ktor/util/CaseInsensitiveMap.kt (hunk 1 above)
Comment (e5l) [reply ktor-utils/common/src/io/ktor/util/CaseInsensitiveMap.kt:44]: It's code style change. Are we ok to keep it here?

Comment (Stexxe) [reply ktor-utils/common/src/io/ktor/util/CaseInsensitiveMap.kt:44]: No

File: ktor-utils/common/src/io/ktor/util/date/GMTDateParser.kt [hunk 8]
```diff
@@ -94,18 +92,19 @@ class GMTDateParser(private val pattern: String) {
         }
     }
 
-    companion object {
-        const val SECONDS = 's'
-        const val MINUTES = 'm'
-        const val HOURS = 'h'
+    @Suppress("KDocMissingDocumentation")
```
Comment (Stexxe) [inline ktor-utils/common/src/io/ktor/util/date/GMTDateParser.kt:95]: Supress

File: ktor-utils/js/src/io/ktor/util/AttributesJs.kt [hunk 9]
```diff
@@ -2,16 +2,18 @@
...
 
+
 package io.ktor.util
 
 /**
  * Create ES specific [Attributes] instance.
  */
 @JsName("AttributesJsFn")
-actual fun Attributes(concurrent: Boolean): Attributes = AttributesJs()
+@Suppress("FunctionName")
```
Comment (Stexxe) [inline ktor-utils/js/src/io/ktor/util/AttributesJs.kt:12]: Suppress

File: ktor-utils/js/src/io/ktor/util/CollectionsJs.kt [hunk 10]
```diff
@@ -4,4 +4,5 @@
 
 package io.ktor.util
 
-actual fun <T> Set<T>.unmodifiable(): Set<T> = this
+@Suppress("KDocMissingDocumentation")
```
Comment (Stexxe) [inline ktor-utils/js/src/io/ktor/util/CollectionsJs.kt:7]: Suppress

File: ktor-utils/js/src/io/ktor/util/CryptoJs.kt [hunk 11]
```diff
@@ -27,8 +27,9 @@ actual fun generateNonce(): String {
 /**
  * Create [Digest] from specified hash [name].
  */
+@Suppress("FunctionName")
```
Comment (Stexxe) [inline ktor-utils/js/src/io/ktor/util/CryptoJs.kt:30]: Suppress

File: ktor-utils/js/src/io/ktor/util/CryptoJs.kt [hunk 12]
```diff
@@ -48,22 +49,23 @@ actual fun Digest(name: String): Digest = object : Digest {
 
 // Variable is renamed to `_crypto` so it wouldn't clash with existing `crypto` variable.
 // JS IR backend doesn't reserve names accessed inside js("") calls
+@Suppress("ObjectPropertyName", "UnsafeCastFromDynamic")
```
Comment (Stexxe) [inline ktor-utils/js/src/io/ktor/util/CryptoJs.kt:52]: Suppress

File: ktor-utils/js/src/io/ktor/util/LockJs.kt [hunk 13]
```diff
@@ -4,11 +4,10 @@
 
 package io.ktor.util
 
-import io.ktor.utils.io.core.*
-
+@Suppress("KDocMissingDocumentation")
```
Comment (Stexxe) [inline ktor-utils/js/src/io/ktor/util/LockJs.kt:7]: Suppress

File: ktor-utils/js/src/io/ktor/util/PlatformUtilsJs.kt [hunk 14]
```diff
@@ -4,19 +4,18 @@
 
 package io.ktor.util
 
-actual object PlatformUtils {
+@Suppress("KDocMissingDocumentation")
```
Comment (Stexxe) [inline ktor-utils/js/src/io/ktor/util/PlatformUtilsJs.kt:7]: Suppress

File: ktor-utils/js/src/io/ktor/util/network/NetworkAddressJs.kt [hunk 15]
```diff
@@ -4,19 +4,40 @@
...
+ * The address will be resolved after construction.
+ *
+ * @throws UnresolvedAddressException if the [hostname] cannot be resolved.
+ */
+public actual fun NetworkAddress(hostname: String, port: Int): NetworkAddress =
+    NetworkAddress(hostname, port, hostname)
 
-actual fun NetworkAddress(hostname: String, port: Int): NetworkAddress =
-    io.ktor.util.network.NetworkAddress(hostname, port, hostname)
+@Suppress("KDocMissingDocumentation")
```
Comment (Stexxe) [inline ktor-utils/js/src/io/ktor/util/network/NetworkAddressJs.kt:42]: Suppress

File: ktor-utils/jvm/src/io/ktor/util/BufferViewJvm.kt [hunk 16]
```diff
@@ -2,6 +2,8 @@
  * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
  */
 
+@file:Suppress("DEPRECATION")
```
Comment (Stexxe) [inline ktor-utils/jvm/src/io/ktor/util/BufferViewJvm.kt:5]: Suppress

**Review by Stexxe — CHANGES_REQUESTED** [review]

File: ktor-utils/common/src/io/ktor/util/StringValues.kt (hunk 7 above)
Comment (e5l) [reply ktor-utils/common/src/io/ktor/util/StringValues.kt:96]: Could you clarify?

**Review by Stexxe — APPROVED** [review]

**Pull request merged**